### PR TITLE
Populate auth form on error

### PR DESCRIPTION
**Pull request by matttarantino**
Keeps the values of form should an error occur so the user doesn't need to fill out everything again

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -18,11 +18,34 @@ app.use(session({
   saveUninitialized: true
 }));
 
+app.use('*', async (req, res, next) => {
+  if (req.session.user) {
+    res.locals.user = true;
+    
+    let user = await users.getUser(req.session.user.username);
```
Comment (dwilliamson5) [inline app.js]: Do we want to catch any exceptions from this

Comment (matttarantino) [reply app.js]: yeah, I can fix that

File: views/auth/signup.handlebars [hunk 2]
```diff
@@ -25,29 +25,34 @@
         <div class="form-text">PASSWORD REQUIREMENTS HERE</div>
     </div>
 
+    <div class="mb-3">
+        <label id="password-confirmation-label" for="password-confirmation" class="form-label">Confirm Password</label>
+        <input id="password-confirmation" type="password" name="password-confirmation" class="form-control" />
+    </div>
```
Comment (dwilliamson5) [inline views/auth/signup.handlebars:31]: Need new issue to track the work for check passwords match

**Review by dwilliamson5 — APPROVED** [review]

File: views/auth/signup.handlebars (hunk 2 above)
Comment (dwilliamson5) [reply views/auth/signup.handlebars:31]: I don't think the name can have a hyphen in it since it's being passed in the req.body. I might be wrong

Comment (matttarantino) [reply views/auth/signup.handlebars:31]: https://github.com/dwilliamson5/CS546_WS_final_project/issues/41

Comment (matttarantino) [reply views/auth/signup.handlebars:31]: Gonna finish reviewing your PR next so that hopefully some of the profile stuff can be copied here

**Pull request merged**